File: model/src/main/java/io/quartic/geojson/Point.java
package io.quartic.geojson;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import io.quartic.common.SweetStyle;
import org.immutables.value.Value;

import java.util.List;

@SweetStyle
@Value.Immutable
@JsonSerialize(as = PointImpl.class)
@JsonDeserialize(as = PointImpl.class)
public interface Point extends Geometry {
    List<Double> coordinates();

    @Override
    default <T> T accept(GeometryVisitor<T> geometryVisitor) {
        return geometryVisitor.visit(this);
    }
}

File: common-core/src/main/kotlin/io/quartic/common/geojson/Point.kt
package io.quartic.common.geojson

import com.fasterxml.jackson.annotation.JsonTypeName

@JsonTypeName("Point")
data class Point @JvmOverloads constructor(
        val coordinates: List<Double>,
        override val crs: Map<String, Any>? = null,
        override val bbox: List<Double>? = null
) : Geometry {
    constructor(vararg coordinates: Double) : this(coordinates.asList())
}
